Add text analytics v3.2-preview.2 API

omarelhariry opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

omar-el-hariry committed:
Add text analytics v3.2-preview.2 API

omarelhariry commented:
@deyaaeldeen

deyaaeldeen reviewed (commented):
Thanks @omarelhariry! I did a quick regeneration using this swagger and my first impression is that it looks good! One thing I noticed though is the absence of the `LoggingOptOut` parameter from the parameters list of the newly added tasks, is that intentional? and why?

kristapratico commented on specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.2/TextAnalytics.json:
```diff
@@ -0,0 +1,3661 @@
+      "type": "array",
+      "uniqueItems": true
+    },
+    "top": {
```
It seems like lines 3145-3351 are new in this version. Just wondering what their purpose is, or if they were added by mistake?

omarelhariry replied:
Yes, they are not needed. Removed them.

omar-el-hariry committed:
CR comments

omarelhariry commented:
@deyaaeldeen No, this was missed out so, I added it. 
Please, take another round of review and tell me if you have other comments.

omarelhariry commented:
@deyaaeldeen @kristapratico Do you have an idea why is build failing? or is this expected?

deyaaeldeen reviewed (approved):
Looks good! The CI error because there is a breaking change between v3.1 and v3.2-preview.2 in the ApiVersion enum. I think this change is ok but it makes me wonder why it is enum in the first place.

deyaaeldeen commented:
@omarelhariry I made a post to merge your PR [here](https://teams.microsoft.com/l/message/19:d6ff4003f5c848a2a2d6dcdfc0ebd497@thread.skype/1629832853098?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1629832853098&teamName=Azure%20SDK&channelName=Service%20-%20Text%20Analytics&createdTime=1629832853098).

omar-el-hariry committed:
Make project and deployment names required parameters

deyaaeldeen commented on specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.2/TextAnalytics.json:
```diff
@@ -0,0 +1,3477 @@
+      "properties": {
+        "parameters": {
+          "properties": {
+            "projectName": {
+              "type": "string"
+            },
+            "deploymentName": {
+              "type": "string"
+            },
```
could you please redefine these two properties as standalone parameters similar to `model-version`? so that you can set their serialized value to be `project-name` and `deployment-name` respectively.

deyaaeldeen replied:
Actually they should be just renamed to `project-name` and `deployment-name`, no need to define them as standalone parameters. For example, look at how `model-version` is defined in other tasks.

kayousef replied:
@deyaaeldeen these names are coming from Language APIs design. You have a good point that we might have them aligning with model-version, but this was not raised in the board review and it's too late now to change this.
This will require a service update and the release have been underway for almost 10 days already.

I'll make sure we have a discussion around this for TA's upcoming Language API and also upcoming previews.

deyaaeldeen replied:
@kayousef actually, the service deployment already respects the proposed change here and gives use `project-name` and `deployment-name` so the change I am requesting is needed for the swagger only to match the deployment responses.

deyaaeldeen replied:
I am marking this thread back as unresolved for now but let me know if that does not work.

kayousef replied:
Fixed

deyaaeldeen reviewed (changes requested):
I left one comment that needs to be fixed before merging.

kayousef committed:
Fix projectName and deploymentName

Rename projectName and deploymentName in input to project-name and deployment-name

deyaaeldeen reviewed (approved):
The swagger looks great! I regenerated with the latest version and everything looks good! Please consider reviewing the examples and make sure they conform to the latest version of the swagger.

abhahn commented on specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.2/TextAnalytics.json:
```diff
@@ -0,0 +1,3477 @@
+      "properties": {
+        "parameters": {
+          "properties": {
+            "project-name": {
```
Just curious why some of the property names are specified using a `-` while others use camel case.

abhahn commented on specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.2/TextAnalytics.json:
```diff
@@ -0,0 +1,3477 @@
+      },
+      "type": "object"
+    },
+    "CustomMultiClassificationTask": {
```
The definition for this is basically the same as the CustomClassificationTask above.  Do you think there may be divergence of these two task types in the future in terms of parameters, or could we theoretically use the same definition for both?
